js/ui/popup.js

```diff
@@ -601,12 +604,45 @@ var Popup = Overlay.inherit({
             this._$bottom && this._$bottom.remove();
             var $bottom = $("<div>").addClass(POPUP_BOTTOM_CLASS).insertAfter(this.$content());
             this._$bottom = this._renderTemplateByType("bottomTemplate", items, $bottom).addClass(POPUP_BOTTOM_CLASS);
+            this.$bottomToolbarContentItems = this._$bottom.find(".dx-toolbar-items-container > div");
             this._toggleClasses();
         } else {
             this._$bottom && this._$bottom.detach();
         }
     },
 
+    _getBottomToolbarWidth: function() {
+        var width = 0;
+
+        if(this.$bottomToolbarContentItems) {
+            this.$bottomToolbarContentItems.each(function(index, item) {
+                width += $(item).outerWidth();
+            });
+        }
+
+        return width;
+    },
+
+    _updateBottomToolbarCompactMode: function() {
+        if(this._$bottom) {
+            this._$bottom.removeClass(POPUP_TOOLBAR_COMPACT_CLASS);
```

Comment:
Use `this._$bottom.toggleClass(POPUP_TOOLBAR_COMPACT_CLASS, this._$bottom.width() < this._getBottomToolbarWidth())`

Reply:
We should remove compact class before full items width calculation